app/rentals_routes.py

```diff
@@ -0,0 +1,112 @@
+from types import ModuleType
+from app import db
+from app.customer_routes import validate_input
+from app.models.video import Video
+from app.models.customer import Customer
+from app.models.rental import Rental
+from flask import Blueprint, jsonify, make_response, request, abort
+import datetime
+from datetime import timedelta, datetime
+
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+# checks out a video to a customer, and updates the data in the database
+@rentals_bp.route("/check-out", methods=["POST"], strict_slashes=False)
+def check_out():
+    request_data = request.get_json()
+    
+    validate_id(request_data)
+
+    video = get_or_return_404(request_data["video_id"], Video)
+    customer = get_or_return_404(request_data["customer_id"], Customer)
+
+    # if the number of videos checked out == the video's total inventory, there are none left to check out
+    if len(video.rentals) == video.total_inventory:
+        return jsonify({"message": "Could not perform checkout"}), 400        
+
+    new_rental = Rental(
+        video_id = request_data["video_id"],
+        customer_id = request_data["customer_id"]
+    )
+    
+    db.session.add(new_rental)
+    db.session.commit()
+
+    return jsonify({
+        "customer_id": new_rental.customer_id,
+        "video_id": new_rental.video_id,
+        "due_date": new_rental.due_date,
+        "videos_checked_out_count": len(customer.videos_rented),
+        "available_inventory": video.total_inventory-len(video.rentals)
+        }), 200
+
+# checks in a video to a customer, and updates the data in the database as such
+@rentals_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+def check_in():
+    request_data = request.get_json()
+    
+    validate_id(request_data)
+
+    video_id = request_data["video_id"]
+    customer_id = request_data["customer_id"]
+
+    video = get_or_return_404(video_id, Video)
+    customer = get_or_return_404(customer_id, Customer)
+    
+    for rental in video.rentals: # every rental of the video currently checked out
+        if rental.customer_id == customer_id:
+            db.session.delete(rental)
+            db.session.commit()
+            return jsonify({
+                "customer_id": customer_id,
+                "video_id": video_id,
+                "videos_checked_out_count": len(customer.videos_rented),
+                "available_inventory": video.total_inventory-len(video.rentals)
```

Comment:
Again I like the calculations here, but I suggest moving this into a method of the `Rental` model.